lemur/plugins/lemur_gcp/plugin.py

```diff
@@ -48,81 +44,108 @@ class GCPDestinationPlugin(DestinationPlugin):
+        {
+            "name": "serviceAccountTokenPath",
+            "type": "str",
+            "required": False,
+            "helpMessage": "Path to vault secret",
+        }
+    ]
 
-        cert_name = f"ssl-{cn}-{authority}-{issued_on}"
+    def get_certificates(self, options, **kwargs):
+        try:
+            credentials = auth.get_gcp_credentials(self, options)
+            project_id = self.get_option("projectID", options)
+            return certificates.fetch_all(project_id, credentials)
+        except Exception as e:
+            current_app.logger.error(
+                f"Issue with fetching certificates from GCP. Action failed with the following log: {e}",
+                exc_info=True,
+            )
+            raise Exception(f"Issue fetching certificates from GCP: {e}")
 
-        return self._modify_cert_name_for_gcp(cert_name)
+    def get_certificate_by_name(self, certificate_name, options):
+        try:
+            credentials = auth.get_gcp_credentials(self, options)
+            project_id = self.get_option("projectID", options)
+            return certificates.fetch_by_name(project_id, credentials, certificate_name)
+        except Exception as e:
+            current_app.logger.error(
+                f"Issue with fetching certificate by name from GCP. Action failed with the following log: {e}",
+                exc_info=True,
+            )
+            raise Exception(f"Issue fetching certificate from GCP: {e}")
 
-    def _modify_cert_name_for_gcp(self, cert_name):
-        # Modify the cert name to comply with GCP naming convention
-        gcp_name = cert_name.replace('.', '-')
-        gcp_name = gcp_name.replace('*', "star")
-        gcp_name = gcp_name.lower()
-        gcp_name = gcp_name[:63]
-        gcp_name = gcp_name.rstrip('.*-')
+    def get_endpoints(self, options, **kwargs):
+        try:
+            credentials = auth.get_gcp_credentials(self, options)
+            project_id = self.get_option("projectID", options)
+            endpoints = fetch_target_proxies(project_id, credentials)
+            return endpoints
+        except Exception as e:
+            current_app.logger.error(
+                f"Issue with fetching endpoints from GCP. Action failed with the following log: {e}",
+                exc_info=True,
+            )
+            raise Exception(f"Issue fetching endpoints from GCP: {e}")
 
-        return gcp_name
+    def update_endpoint(self, endpoint, certificate):
+        options = endpoint.source.options
+        credentials = auth.get_gcp_credentials(self, options)
+        project_id = self.get_option("projectID", options)
+        update_target_proxy_cert(project_id, credentials, endpoint, certificate)
 
-    def _full_ca(self, body, cert_chain):
-        # in GCP you need to assemble the cert body and the cert chain in the same parameter
-        return f"{body}\n{cert_chain}"
+    def clean(self, certificate, options, **kwargs):
+        raise NotImplementedError
```

Comment:
Nitpick. This looks like the same method and logic that is on the Parent class. I think we should just remove this method from here and rely on `SourcePlugin` to handle this.